### PR TITLE
Added docs for option procs

### LINKED ISSUE #154: [FeatureRequest] Add convenience procs to check if a formparam/queryparam exists
Right now, prologue has getQueryParam / getPathParam / getPostParam / getFormParam.
However, for my own usecases I have scenarios where the same controller handles multiple possible URLs / form input possibilities and I'd like to be able to just check whether a given request *has* a specific form param/pathParam/queryParam etc.

So convenience procs like "hasQueryParam" etc. would be very useful to me. Would it be desired to add such procs to prologue? Alternatively, the same could be achieved by having the "get" procs above return Option types, to symbolize that a given param might not exist.

### PR DESCRIPTION
**Pull request by PhilippMDoerner**
close https://github.com/planety/prologue/issues/154

.... I am honestly not sure why the heck this is showing up as 11 commits. That's 1 commit for the changes in docs and the other 10 are ones that have already been merged into prologue devel in the last PR... mysteries of github?

### CODE REVIEW

File: docs/context.md [hunk 1]
```diff
@@ -13,36 +13,42 @@ proc login*(ctx: Context) {.async.} =
...
+Gets the query strings(for example, "www.google.com/hello?name=12", `name=12`) as an `Option[string]`.
 
 ```nim
 proc hello(ctx: Context) {.async.} =
-  doAssert ctx.getQueryParams("name") == "12"
+  doAssert ctx.getQueryParamsOption("name").get() == "12"
 ```
 
-### getPathParams
+If you want to specify a default value to return, take a look at <a href="https://planety.github.io/prologue/coreapi/context.html#getQueryParams%2CContext%2Cstring%2Cstring"> getQueryParams </a>
```
Comment (ringabout) [inline docs/context.md]: How about using [getQueryParams](https://planety.github.io/prologue/coreapi/context.html#getQueryParams%2CContext%2Cstring%2Cstring)
```
[getQueryParams](https://planety.github.io/prologue/coreapi/context.html#getQueryParams%2CContext%2Cstring%2Cstring)
```

Comment (PhilippMDoerner) [reply docs/context.md]: I am very much not knowledgeable in the ways of writing proper markdown, I know how to do headers. I'll adjust the links.

Comment (PhilippMDoerner) [reply docs/context.md]: Done

**Review by ringabout — APPROVED** [review]
thank you!

**Comment by xflywind** [comment]
close https://github.com/planety/prologue/issues/154

**Comment by ringabout** [comment]
> I am honestly not sure why the heck this is showing up as 11 commits. That's 1 commit for the changes in docs and the other 10 are ones that have already been merged into prologue devel in the last PR... mysteries of github

You can use git rebase to adjust commit messages. But it is unnecessary because Prologue uses squash merge.

**Pull request merged**